packages/cli-hydrogen/src/cli/services/build.ts

```diff
@@ -10,46 +10,46 @@ interface DevOptions {
   assetBaseURL?: string
 }
 
-async function build({directory, targets, base, assetBaseURL}: DevOptions) {
+export function buildTaskList({directory, targets, base, assetBaseURL}: DevOptions): ui.ListrTask[] {
   const commonConfig = {base, root: directory}
 
-  const tasks: ui.ListrTask[] = Object.entries(targets)
+  return Object.entries(targets)
     .filter(([_, value]) => value)
-    .map(([key, value]) => {
-      return {
-        title: `Building ${key} code`,
-        task: async (_, task) => {
-          if (key === 'worker') {
-            process.env.WORKER = 'true'
-          }
-          if (assetBaseURL) {
-            process.env.HYDROGEN_ASSET_BASE_URL = assetBaseURL
-          }
-
-          try {
-            await viteBuild({
-              ...commonConfig,
-              build: {
-                outDir: `dist/${key}`,
-                ssr: typeof value === 'string' ? value : undefined,
-                manifest: key === 'client',
-              },
-            })
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          } catch (error: any) {
-            const abortError = new kitError.Abort(error.message)
-            abortError.stack = error.stack
-            throw abortError
-          }
-
-          task.title = `Built ${key} code`
-        },
-      }
-    })
+    .map(([key, value]) => ({
+      title: `Building ${key} code`,
+      task: async (_, task) => {
+        if (key === 'worker') {
+          process.env.WORKER = 'true'
+        }
+        if (assetBaseURL) {
+          process.env.HYDROGEN_ASSET_BASE_URL = assetBaseURL
+        }
+
+        try {
+          await viteBuild({
+            ...commonConfig,
+            build: {
+              outDir: `dist/${key}`,
+              ssr: typeof value === 'string' ? value : undefined,
+              manifest: key === 'client',
+            },
+          })
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
```

Comment:
Fully agree with that and the article is very useful will keep in mind for the future. This code was already written so preferred to make my changes as minimal as possible. Everything is green just because I split the code.